src/App.tsx

```diff
@@ -1,26 +1,78 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
 import logo from './logo.svg';
 import './App.css';
+import { BATCH_DUMP_TIMEOUT_MS, FilesApi } from './api-client';
+
+const fileIds = ['fileid1', 'fileid2', 'fileid3'];
+
+const App = () => {
+  let combinedFiles: string[] | undefined;
+  const [files, setFiles] = useState<string[]>([]);
+  const getFiles = useCallback(FilesApi.getClientWithBatching(), []);
+
+  const [batchInProgress, setBatchInProgress] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const requestFile = useCallback(
+    async (ids: string[]) => {
+      try {
+        setBatchInProgress(true);
+        const { data } = await getFiles(ids);
+        const newFiles = data.items.map(file => file.id);
+
+        if (!combinedFiles) {
+          combinedFiles = [];
+        }
+        combinedFiles = combinedFiles.concat(newFiles);
+        setFiles(Array.from(new Set(combinedFiles)));
+
+        setBatchInProgress(false);
+        setErrorMessage('');
+      } catch (e: any) {
+        setBatchInProgress(false);
+        setErrorMessage(e.message);
+      }
+    },
+    [files]
+  );
 
-function App() {
   return (
     <div className="App">
       <header className="App-header">
         <img src={logo} className="App-logo" alt="logo" />
+        <h2>Let&apos;s make some batch requests</h2>
```

Comment:
This could have probably been split into a couple of components. But they would have been quite small and wouldn't have had much logic. So I just did it all in one file.